tournament/src/api/views/tournament_matches_views.py

```diff
@@ -0,0 +1,153 @@
+from tournament.authenticate_request import authenticate_request
+from tournament import settings
+from api import error_message as error
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class GenerateMatchesView(View):
+    @staticmethod
+    def post(request: HttpRequest, tournament_id: int) -> JsonResponse:
+        user, authenticate_errors = authenticate_request(request)
+        if user is None:
+            return JsonResponse(data={'errors': authenticate_errors}, status=401)
+
+        try:
+            body = json.loads(request.body.decode('utf8'))
+        except json.JSONDecodeError:
+            return JsonResponse(data={'errors': [error.BAD_JSON_FORMAT]}, status=400)
+
+        try:
+            tournament = Tournament.objects.get(id=tournament_id)
+            players = list(tournament.players.all())
+            GenerateMatchesView.set_players_elo(players)
+            tournament.matches.all().delete()
+        except ObjectDoesNotExist:
+            return JsonResponse({'errors': [f'tournament with id `{tournament_id}` does not exist']}, status=404)
+        except Exception as e:
+            return JsonResponse({'errors': [str(e)]}, status=500)
+
+        error_message, status_code = GenerateMatchesView.error_handler(user['id'], tournament, players)
+        if error_message is not None:
+            return JsonResponse({'errors': [error_message]}, status=status_code)
+
+        players = GenerateMatchesView.sort_players(players, body.get('is_random', False))
+        matches = GenerateMatchesView.generate_matches(players, tournament)
+
+        try:
+            Match.objects.bulk_create(matches)
+        except Exception as e:
+            return JsonResponse({'errors': [str(e)]}, status=500)
+
+        return JsonResponse(GenerateMatchesView.matches_to_json(matches), status=200)
+
+    @staticmethod
+    def sort_players(players: list[Player], is_random: any) -> list[Player]:
+        if isinstance(is_random, bool) and is_random:
+            random.shuffle(players)
+        else:
+            players.sort(key=lambda x: x.elo, reverse=True)
+        return players
+
+    @staticmethod
+    def generate_matches(players: list[Player], tournament: Tournament) -> list[Match]:
+        nb_players = len(players)
+        seed_order = GenerateMatchesView.get_seed_order(nb_players)
+        matches = []
+        i = 0
+
+        for match in seed_order:
+            if match[0] - 1 < nb_players:
+                player_1 = players[match[0] - 1]
+            else:
+                player_1 = None
+            if match[1] - 1 < nb_players:
+                player_2 = players[match[1] - 1]
+            else:
+                player_2 = None
+            matches.append(
+                Match(
+                    player_1=player_1,
+                    player_2=player_2,
+                    tournament=tournament,
+                    match_id=i
+                )
+            )
+            i += 1
+
```

Comment:
You can use the enumerate function if you want to iterate and obtain indices

Reply:
c3889f8eddc5ada20e9998a0d62b5fe40918dca3